extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergRefreshingTableLocationProvider.java

```diff
@@ -0,0 +1,107 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.layout;
+
+import io.deephaven.engine.table.impl.locations.*;
+import io.deephaven.engine.table.impl.locations.impl.AbstractTableLocationProvider;
+import io.deephaven.engine.table.impl.locations.impl.TableLocationFactory;
+import io.deephaven.engine.table.impl.locations.impl.TableLocationKeyFinder;
+import io.deephaven.engine.table.impl.locations.util.TableDataRefreshService;
+import org.apache.iceberg.Snapshot;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Polling-driven {@link TableLocationProvider} implementation that delegates {@link TableLocationKey location key}
+ * discovery to a {@link TableLocationKeyFinder} and {@link TableLocation location} creation to a
+ * {@link TableLocationFactory}.
+ */
+public class IcebergRefreshingTableLocationProvider<TK extends TableKey, TLK extends TableLocationKey>
```

Comment:
This class seems to be conflating the auto and manual implementations a bit. I'm not reviewing in great detail, as I think it's still a WIP.